packages/react-pose/src/components/Transition/children.ts

```diff
@@ -86,7 +93,7 @@ const mergeChildren = ({
       onPoseComplete: removeFromTree(key),
       popFromFlow: flipMove,
       ...propsForChild,
-      ...filterChildProps(child.props)
+      ...filterChildProps(child.props, Object.keys(propsForChild))
```

Comment:
this approach is IMHO faulty for edge cases, it depends though what kind of edge cases do you want to support

a possible situation is that we first render a child with no prop A inside a Transition with that prop A, the prop gets merged onto this child, later we rerender and the child receives a prop A this time (totally possible, although imho a code smell), the new child prop A should "win" but instead you filter it out here

also the problem here might be that `propsForChild` are different between rerenders and this might lead to buggy behaviours

in this exact scenario my first point is not so valid because this branch handles only leaving nodes, but I'm wondering if the same bug doesn't also affect any other situation? doesn't it also affect moving elements?